### DISCUSSION TITLE
Set value to model field outside the form in a Resource

### DISCUSSION THREAD

**Discussion by francisnnumbi** [filamentphp/filament#8760] [Help] [answered]
Hey,

From the Resource, in the `static function form()`, we can build inputs and this form is returned. I find myself in the situation where there is a value that I need to set to the model before it is saved to database. Eg. I need to store the id of the logged in user who has added or modified a record. This should be added behind the scene before the record is saved. How can I do it ?

**Comment by francisnnumbi** [comment]
I have found a way. I don't know it is a good one. 
Inside form() function, before returning $form, I did  `$form->getRecord()->user_id  = Auth::id().'';`. I called getRecord from $form, fill in the fields I want, then I return $form.

However, this technic only works on existing record, when updating. It doesn't work on new record.

**Reply by leandrocfe** [reply]
[customizing-data-before-saving](https://filamentphp.com/docs/3.x/panels/resources/creating-records#customizing-data-before-saving) ?

**Reply by francisnnumbi** [reply]
Thank you very much

**Comment by francisnnumbi** [comment]
Thank you very much. It worked well on the page, but I don't understand when it comes to modals. This documentation [here](https://filamentphp.com/docs/3.x/actions/prebuilt-actions/create#customizing-data-before-saving) doesn't clarify where exactly that method or that line is to be put. I have read several times, but the snipet is not clear and doesn't give a hint.

**Reply by francisnnumbi** [reply]
@leandrocfe can you please help ?

**Reply by leandrocfe** [reply]
[here](https://filamentphp.com/docs/3.x/actions/prebuilt-actions/create#customizing-data-before-saving)

**Reply by francisnnumbi** [reply]
Actually @leandrocfe , I saw it. But my issue is where exactly I have to write it ? inside form() method ? inside table() method ? inside what method exactly ? it is not stated in the documentation !

**Reply by leandrocfe** [reply, marked as answer]
If you have a resource `simple` you should add on the ManageXX.php, because you have a `createAction` action like

```php
protected function getHeaderActions(): array
{
    return [
        CreateAction::make()
            ->mutateFormDataUsing(function (array $data): array {
                $data['user_id'] = auth()->id();

                return $data;
            })
    ];
}
```

**Reply by francisnnumbi** [reply]
Oh yeah. I missed the point with the file ManageXX. It has worked. Thank you very much. I think that file should be mentioned in the documentation. It will help noobs like me !
